frontend/packages/dev-console/src/components/topology/componentUtils.ts

```diff
@@ -249,18 +258,26 @@ const createConnectorCallback = (serviceBinding: boolean) => (
   source: Node,
   target: Node | Graph,
 ): React.ReactElement[] | null => {
+  if (source === target) {
+    return null;
+  }
+
   if (isGraph(target)) {
     return graphContextMenu(target, source);
   }
   if (target.isGroup()) {
     return groupContextMenu(target, source);
   }
-  createConnection(source, target, null, serviceBinding);
+  createConnection(source, target, null, serviceBinding).catch((error) => {
+    errorModal({ title: 'Error creating connection', error: error.message, showIcon: true });
+  });
   return null;
 };
 
 const removeConnectorCallback = (edge: Edge): void => {
-  removeConnection(edge);
+  removeConnection(edge).catch((error) => {
+    errorModal({ title: 'Error removing connection', error: error.message });
```

Comment:
Why no icon here but there's an icon in the other error?

Reply:
oversight, I will update